code/__HELPERS/unsorted.dm

```diff
@@ -597,12 +597,14 @@ Returns: A list of all areas of that type in the world.
 		for(var/V in GLOB.sortedAreas)
 			var/area/A = V
 			if(cache[A.type])
-				areas += V
+				if(target_z == 0 || A.z == target_z)
```

Comment:
You are right, it doesn't work that way. I'll redo some things.

Reply:
Actually no, I redid my tests second time, correcting my error and upon placing a few door in `/area/maint` in CC z level and activating emergency maint access (with added station z level argument) it didn't set those to emergency access,
I redid the test for all `/area` (in case I got some wrong maint) and they didn't turn emergency.

Reply:
@PowerfulBacon